transactions/README.md

```diff
@@ -234,12 +236,26 @@ Commit
 ```
 
 ## ARIES Data Structures
+### Stable Storage
+* A **log** composed of records, which contain a log sequence number (monotonically increasing), as well as a pointer to the previous log record for the same transaction.  
+* **Page**, with a page log sequence number, which uniquely identifies the log record for the latest update applied to this page.
 
-TODO: Summarize L15 slides 43-46
+### In-Memory
+* Transaction table (**T-table**) tracks transactions and their status (running, committed, aborted), as well as the most recent log record written by the transaction.  
+* Dirty page table: a page identifer table for dirty pages. It uses a recovery log sequence number, which tracks the earliest change to that page that is *not on disk*
 
 ## ARIES Crash Recovery Trace
+### Phase 1 (Analysis)
+1. Start at **checkpointed** entries in the T-table and dirty page table.
+2. Read the log forward form the checkpoint, and update tables.
+3. Figure out which transactions committed since the checkpoing, and which failed.
+
+### Phase 2 (Redo)
+1. Start at the first log sequence number and scan log entries forward in time. Reapply **all actions** and update the page log sequence numbers.
+2. State now matches state as recorded by log.
```

Comment:
This isn't really an operation, it's just an observation. Maybe you could rm this and just detail 1 a bit more?